### PR TITLE
CI Infra

### PR DESCRIPTION
**Pull request by tewaro**
Initial ci attempt

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+        fi
+        if [ ! -e `which makers` ]; then
+          cargo install -j cargo-make
+        fi
+
+    - name: Docker Build
+      timeout-minutes: 10
+      run: |
+        $HOME/.cargo/bin/cargo make dockerbuild
```
Comment (patrickkenney9801) [inline .github/workflows/build-and-test.yml:53]: why create a docker image?

Comment (tewaro) [reply .github/workflows/build-and-test.yml:53]: Usually this is super fast cause the image does not change. But this way there is a single simple workflow that is repeatable even without linux.

Comment (patrickkenney9801) [reply .github/workflows/build-and-test.yml:53]: so this is just a test that the Dockerfile is syntactically correct?

**Review by meyerzinn — APPROVED** [review]

**Review by patrickkenney9801 — APPROVED** [review]

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (tewaro) [reply .github/workflows/build-and-test.yml:53]: it also gives it enough time to build if the container has changed.

**Pull request merged**